### ISSUE TITLE
Invalid #pragma parsing

### ISSUE THREAD

**Issue by AlexBAV** [boostorg/wave#206]
After #188 the following source code is parsed incorrectly by Boost.Wave:

```c++
#pragma some(some)
#include "path"
```

The line end at the end of pragma line is "eaten" and next token after `right paren` returned is literally `#include "path"`. Adding another newline after the `#pragma` line fixes this.

This started in boost 1.84.0, apparently after #188.

**Comment by AlexBAV** [comment]
Sorry for reporting, duplicate of #200.

**Comment by jefftrull** [comment]
Sorry about the problem, hope you are able to use our develop branch until 1.85 is released.
